core/src/Slay/Prim.hs

```diff
@@ -68,12 +70,14 @@ deriving instance (Show (g (Maybe Natural)), Show (g (Color))) => Show (PrimText
 -- from -1 to 1
 newtype Curvature = Curvature Rational
 
+-- | curve direction
 data Direction =
   Direction
     { directionLeftToRight :: Bool,
       directionTopToBottom :: Bool
     }
 
+-- | adt for curves
```

Comment:
This comment is redundant because it's just restating what the code does in English. Generally, I expect comments to explain *why*, not *what*.